### DISCUSSION TITLE
For sorting of tailwindcss classes

### DISCUSSION THREAD

**Discussion by Himanshu-Sorathiya** [tailwindlabs/tailwindcss#14248] [Help]
Is there any vscode extention or plugins there to sort tailwindcss classes automatically, which sorts this line
<div class="sm:p-6 sm:bg-blue-500 lg:bg-yellow-500 dark:bg-gray-800 hover:bg-red-700 focus:p-12 md:m-8 lg:m-10 sm:text-left md:text-right lg:text-justify dark:text-gray-300 p-4 m-4 text-center text-gray-900 bg-red-500">

to this ( or near similer )
<div
  class="m-4 md:m-8  lg:m-10 bg-red-500 sm:bg-blue-500 lg:bg-yellow-500  hover:bg-red-700 dark:bg-gray-800 p-4  sm:p-6 focus:p-12  text-center  sm:text-left  md:text-right       lg:text-justify text-gray-900  dark:text-gray-300"
>

varients of classes should be near to its og class, unlike prettier tcss plugin which sorts it to end,
unknown class to start
and also it support @apply in css and in other files js/ts

**Comment by tembbo** [comment]
Have you read this: https://tailwindcss.com/blog/automatic-class-sorting-with-prettier

**Reply by Himanshu-Sorathiya** [reply]
yes i read that, its very good but it sorts every varients like sm: md: lg: hover: focus: active: dark: to the end, i want them to be with its og class, its easier for me to find and understand

**Reply by tembbo** [reply]
ok, sorry, but i dont know that much about tailwind (like i just use it for the simplicity). good luck solving it though

**Comment by tembbo** [comment]
Sorry for pinging, but there could be something in the readme here: https://github.com/tailwindlabs/prettier-plugin-tailwindcss/blob/main/README.md
